scripts/keys.coffee

```diff
@@ -22,81 +22,213 @@
+				msg.send "That's utter lies! How can you blame a bot to have the keys?"
 			else if othername is robot.name
-				msg.send "How am I supposed to take those keys? #{name} is a liar!"
+				msg.send "That's utter lies! How can you blame a bot to have the keys?"
 			else
 				users = robot.brain.userForName othername
 				if users is null
 					msg.send "I don't know anyone by the name #{othername}"
 				else
 					k = users.name
-					msg.send "Okay, so now the keys are with #{users.name}"	
-
-		robot.brain.set("key",k)			
-
-	robot.respond /(i|I) (have given|gave|had given) (the key|key|keys|a key|the keys) to (.+)/i , (msg)->
-		othername = msg.match[4]
-		name = msg.message.user.name
-		k = key()
-		if othername is 'you'
-			msg.send "That's utter lies! How can you blame a bot to have the keys?"
-		else if othername is robot.name
-			msg.send "That's utter lies! How can you blame a bot to have the keys?"
-		else
-			users = robot.brain.userForName othername
-			if users is null
-				msg.send "I don't know anyone by the name #{othername}"
+					msg.send "Okay, so now the ravindra keys are with #{users.name}"
+			robot.brain.set("ravindra_key",k)	
+		else if keyname is 'kb' || keyname is 'kasturba'
+			k = kb_key()
+			if othername is 'you'
+				msg.send "That's utter lies! How can you blame a bot to have the keys?"
+			else if othername is robot.name
+				msg.send "That's utter lies! How can you blame a bot to have the keys?"
 			else
-				k = users.name
-				msg.send "Okay, so now the keys are with #{users.name}"	
-			
-
-		robot.brain.set("key",k)		
-				
-	robot.respond /(who|who all) (has|have) (the key|key|keys|a key)/i , (msg)->
-		k = key()
-		msgText = k
-		if msgText is ""
-			msg.send "Ah! Nobody informed me about the keys. Don't hold me responsible for this :expressionless:"
-		else
-			msgText+=" has keys"
-			msg.send msgText	
-		robot.brain.set("key" ,k)	
+				users = robot.brain.userForName othername
+				if users is null
+					msg.send "I don't know anyone by the name #{othername}"
+				else
+					k = users.name
+					msg.send "Okay, so now the kb keys are with #{users.name}"
+			robot.brain.set("kb_key",k)
+		else if keyname is 'master'
+			k = master_key()
+			if othername is 'you'
+				msg.send "That's utter lies! How can you blame a bot to have the keys?"
+			else if othername is robot.name
+				msg.send "That's utter lies! How can you blame a bot to have the keys?"
+			else
+				users = robot.brain.userForName othername
+				if users is null
+					msg.send "I don't know anyone by the name #{othername}"
+				else
+					k = users.name
+					msg.send "Okay, so now the master keys are with #{users.name}"	
+			robot.brain.set("master_key",k)		
+		
+#this section is to print the details about the key holder.		
+	robot.respond /(who|who all) (has|have|has the|have the|has a|have a) (.+) (key|keys)/i , (msg)->
+		othername = msg.match[3]
+		if othername is 'ravindra' || othername is 'rv'
+			k1 = ravindra_key()
```

Comment:
Prefer better names than `k1`, `k2` and `k3` :P 
Also, the code block below can optimised too

Reply:
@IronicallySerious , @hritvi !! please review the changes..